### PR TITLE
Stop using `TokenList` and `NonEmptyTokenList`

### PR DESCRIPTION
**Pull request by ahoppen**
`TokenList` and `NonEmptyTokenList` are anti-patterns for SwiftSyntax parsing because they provide no structure. Stop using them.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-format/pull/476

@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-format/pull/476

@swift-ci Please test

File: gyb_syntax_support/ExprNodes.py [hunk 1]
```diff
@@ -374,13 +374,19 @@
              Child('Type', kind='Type'),
          ]),
 
+    Node('ClosureCaptureItemSpecifier', name_for_diagnostics='closure capture specifier', kind='Syntax',
```
Comment (bnbarham) [inline gyb_syntax_support/ExprNodes.py:377]: Are there existing tests for this? If not it would be good to add a few.

**Review by bnbarham — APPROVED** [review]

File: gyb_syntax_support/ExprNodes.py (hunk 1 above)
Comment (ahoppen) [reply gyb_syntax_support/ExprNodes.py:377]: We’ve got e.g. 
https://github.com/apple/swift-syntax/blob/5545afb01ce3d8e32b1e7b1d4390001625f08d42/Tests/SwiftParserTest/ExpressionTests.swift#L48-L54

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
https://github.com/apple/swift-format/pull/476

@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-format/pull/476

@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-format/pull/476

@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-format/pull/476
https://github.com/apple/swift-stress-tester/pull/223

@swift-ci Please test

**Pull request merged**